### PR TITLE
added: warning for duplicate nodenames

### PR DESCRIPTION
**Pull request by gushil**
#40

### CODE REVIEW

**Comment by gushil** [comment]
@MartijnR 
I don't know why appveyor build is always failed.

**Comment by gushil** [comment]
Sorry, my bad for not doing confirmation first.

PR is updated.

**Comment by MartijnR** [comment]
> I don't know why appveyor build is always failed.

It's building a Windows executable to be used for pyxform Windows users. It was actually initially meant to be used by OpenClinica, but OC decided to split validation from pyxform (and no need to run on Windows). So now, we're just waiting for somebody to complain before fixing this (I don't think anybody is inconvenienced by it).

File: src/xform.js [hunk 1]
```diff
@@ -299,6 +299,20 @@ class XForm {
...
+
+            // add warning for duplicate nodenames
+            const dataEl = children[0];
+            const dataNodes = dataEl.querySelectorAll( '*' );
+            if ( dataNodes.length > 0 ) {
+                let dataNodeNames = [];
+                dataNodes.forEach( dataNode => {
+                    dataNodeNames.push(dataNode.nodeName);
+                } );
+                const uniqueDataNodeNames = new Set( dataNodeNames );
```
Comment (MartijnR) [inline src/xform.js]: Cool! I learned something new! (Set).

**Pull request merged**